src/common/utility/tarray.h

```diff
@@ -453,6 +453,8 @@ class TArray
 
 	void Delete (unsigned int index, int deletecount)
 	{
+		if(index >= Count) return;
```

Comment:
I could be missing something, but I don’t see where this is necessary.

Reply:
As for the message, it's triggered when the array has, ex. 10 values and you're trying to delete past that, ex. `Delete(15,2)`, though yeah, i should've updated the wording when the count check was removed

Reply:
> This isn't needed for ZScript, but it serves as a check to prevent native code that uses `TArray::Delete` from possibly corrupting memory.
> 
> The single-delete function does that check as well (though inverted):
> 
> ```c++
> void Delete (unsigned int index)
> {
> 	if (index < Count)
> 	{
> ```

No, I get that. But it’s covered in it’s entirety by later checks :P

Reply:
it is not, the later cases only check/modify deletecount (this exact failure was causing gzdoom's memory to be corrupted before this)

Reply:
https://user-images.githubusercontent.com/15188209/214144544-61ec48a7-0ae5-40b1-bb07-c1687081d9c6.mp4

Reply:
Ahhhh you meant integer overflow not array overflow ;)  Yeah no, that makes sense to me :)